### PR TITLE
stdlib: import defun.kl, export from prelude at p0

### PR DESCRIPTION
**Pull request by langston-barrett**
This one seems useful enough to put in the prelude.

### CODE REVIEW

**Comment by langston-barrett** [comment]
I don't understand the problem:
```
      user error
    pmatch:                                            Ambiguous identifier in phase p0 #[defun.kl:13.33-13.38]<nflet>
    ScopeSet {_universalScopes = fromList [Scope {scopeNum = 0, scopePurpose = "Root for phase p0"},Scope {scopeNum = 13, scopePurpose = "Module root for /home/travis/build/gelisam/klister/stdlib/defun.kl"},Scope {scopeNum = 68, scopePurpose = "For import at b/defun.kl:2.1-3.1"}], _phaseScopes = fromList [(Phase {phaseNum = 0},fromList [Scope {scopeNum = 0, scopePurpose = "Root for phase p0"}])]}
    ScopeSet {_universalScopes = fromList [Scope {scopeNum = 0, scopePurpose = "Root for phase p0"},Scope {scopeNum = 13, scopePurpose = "Module root for /home/travis/build/gelisam/klister/stdlib/defun.kl"},Scope {scopeNum = 68, scopePurpose = "For import at b/defun.kl:2.1-3.1"}], _phaseScopes = fromList [(Phase {phaseNum = 0},fromList [Scope {scopeNum = 0, scopePurpose = "Root for phase p0"}])]}
    ScopeSet {_universalScopes = fromList [Scope {scopeNum = 0, scopePurpose = "Root for phase p0"},Scope {scopeNum = 13, scopePurpose = "Module root for /home/travis/build/gelisam/klister/stdlib/defun.kl"},Scope {scopeNum = 68, scopePurpose = "For import at b/defun.kl:2.1-3.1"}], _phaseScopes = fromList [(Phase {phaseNum = 0},fromList [Scope {scopeNum = 0, scopePurpose = "Root for phase p0"}])]}
    ScopeSet {_universalScopes = fromList [Scope {scopeNum = 0, scopePurpose = "Root for phase p0"},Scope {scopeNum = 13, scopePurpose = "Module root for /home/travis/build/gelisam/klister/stdlib/defun.kl"},Scope {scopeNum = 68, scopePurpose = "For import at b/defun.kl:2.1-3.1"}], _phaseScopes = fromList [(Phase {phaseNum = 1},fromList [Scope {scopeNum = 0, scopePurpose = "Root for phase p0"}])]}
FAIL
```

**Comment by gelisam** [comment]
"Ambiguous identifier" means there are multiple scope-sets which seem equally good, neither shadows all the others. I am guessing it's because `nflet` is only available at phase 0, not phase 1, but I'd have to think much harder in order to explain how that results in multiple scope-sets seeming equally good, rather than in zero variables of that name being found.

File: examples/defun.kl [hunk 1]
```diff
@@ -1,22 +1,6 @@
 #lang "prelude.kl"
 [import [shift "prelude.kl" 1]]
```
Comment (gelisam) [inline examples/defun.kl:2]: we're not defining any macros, so the shifted import is no longer necessary.

Comment (langston-barrett) [reply examples/defun.kl:2]: Nice catch!

**Comment by gelisam** [comment]
I now think that on the contrary, `nflet` _should_ only be available at phase 0, not phase 1, because it is part of the code which the macro manipulates, not part of the code which defines the macro. Maybe the `rename` bit is buggy?

**Comment by david-christiansen** [comment]
I'm focusing on the abstract this morning, but I'll take a peek as soon as I can find time!

**Comment by david-christiansen** [comment]
We expect this to pass tests after #80 is merged, right?

**Comment by langston-barrett** [comment]
> We expect this to pass tests after #80 is merged, right?

I hope so!

**Pull request merged**